segfault in p2p stream handler while gracefully shutting down beacon chain

# 🐞 Bug Report

### Description

When turning off with a SIGINT, my beacon-chain segfaulted while shutting down. I was running with a lot of peers and it looks like it segfaulted in some p2p handler, but what do I know XD

### Has this worked before in a previous version?

I have not seen this error before.

## 🔬 Minimal Reproduction
Beacon chain points to local geth node.
`geth --goerli --rpc --port 30305 --rpcport 8555`
`./prysm.sh beacon-chain --http-web3provider=http://localhost:8555 --p2p-max-peers 256`

## 🔥 Error

<pre><code>
```[2020-08-05 19:23:15]  INFO p2p: Peer connected activePeers=136 direction=Outbound multiAddr=/ip4/50.101.139.111/tcp/9000/p2p/16Uiu2HAmGkY2kkDgbrQSAtwMJMC1NhYQsfVobiVyLjo7RQ4VENkA
[2020-08-05 19:23:20]  INFO blockchain: Synced new block block=0x48d326d3... epoch=322 finalizedEpoch=320 finalizedRoot=0xd5ae941d... slot=10316
[2020-08-05 19:23:20]  INFO blockchain: Finished applying state transition attestations=74 attesterSlashings=0 deposits=0 proposerSlashings=0 voluntaryExits=0
^C[2020-08-05 19:23:22]  INFO node: Got interrupt, shutting down...
[2020-08-05 19:23:22]  INFO node: Stopping beacon node
[2020-08-05 19:23:22]  WARN sync: Subscription next failed error=context canceled topic=/eth2/e7a75d5a/beacon_block/ssz_snappy
[2020-08-05 19:23:22]  WARN sync: Subscription next failed error=context canceled topic=/eth2/e7a75d5a/beacon_aggregate_and_proof/ssz_snappy
[2020-08-05 19:23:22]  WARN sync: Subscription next failed error=context canceled topic=/eth2/e7a75d5a/beacon_attestation_12/ssz_snappy
[2020-08-05 19:23:22]  WARN sync: Subscription next failed error=context canceled topic=/eth2/e7a75d5a/beacon_attestation_38/ssz_snappy
[2020-08-05 19:23:22]  WARN sync: Subscription next failed error=context canceled topic=/eth2/e7a75d5a/proposer_slashing/ssz_snappy
[2020-08-05 19:23:22]  WARN sync: Subscription next failed error=context canceled topic=/eth2/e7a75d5a/beacon_attestation_4/ssz_snappy
[2020-08-05 19:23:22]  WARN sync: Subscription next failed error=context canceled topic=/eth2/e7a75d5a/attester_slashing/ssz_snappy
[2020-08-05 19:23:22]  WARN sync: Subscription next failed error=context canceled topic=/eth2/e7a75d5a/voluntary_exit/ssz_snappy
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x132ceed]

goroutine 36085398 [running]:
github.com/prysmaticlabs/prysm/beacon-chain/sync.(*limiter).validateRequest(0x0, 0x1ca03a0, 0xc0072d2960, 0x1, 0x0, 0x0)
	beacon-chain/sync/rate_limiter.go:66 +0x6d
github.com/prysmaticlabs/prysm/beacon-chain/sync.(*Service).pingHandler(0xc0003f0a00, 0x1c87b00, 0xc02f58fbf0, 0x15916a0, 0xc024eec3c8, 0x1ca03a0, 0xc0072d2960, 0x0, 0x0)
	beacon-chain/sync/rpc_ping.go:27 +0xb9
github.com/prysmaticlabs/prysm/beacon-chain/sync.(*Service).registerRPC.func1(0x1ca03a0, 0xc0072d2960)
	beacon-chain/sync/rpc.go:124 +0xeb0
github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).SetStreamHandler.func1(0xc01c26bb00, 0x28, 0x7fe134263fe8, 0xc0072d2960, 0x1, 0x0)
	external/com_github_libp2p_go_libp2p/p2p/host/basic/basic_host.go:566 +0x9d
created by github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).newStreamHandler
	external/com_github_libp2p_go_libp2p/p2p/host/basic/basic_host.go:416 +0x63c
```
</code></pre>


## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
```Linux version 5.4.0-42-generic (buildd@lgw01-amd64-038) (gcc version 9.3.0 (Ubuntu 9.3.0-10ubuntu2)) #46-Ubuntu SMP Fri Jul 10 00:24:02 UTC 2020```
  </code>
</pre>

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
`beacon-chain-v1.0.0-alpha.18-linux-amd64`
  </code>
</pre>

**Anything else relevant (validator index / public key)?**
Beacon chain was running for ~34 hrs.

--- a/beacon-chain/sync/service.go
+++ b/beacon-chain/sync/service.go
@@ -166,7 +166,6 @@ func (s *Service) Stop() error {
 	defer func() {
 		if s.rateLimiter != nil {
 			s.rateLimiter.free()
-			s.rateLimiter = newRateLimiter(s.p2p)
 		}
 	}()
 	defer s.cancel()

